ecl/hqlcpp/hqlcse.cpp

```diff
@@ -106,6 +106,7 @@ bool canCreateTemporary(IHqlExpression * expr)
     case no_thisnode:
     case no_libraryscopeinstance:
     case no_loopbody:
+    case no_external:
```

Comment:
This fixes csvheader4.ecl which was broken, but I hadn't picked up on the regression suite differences.  Possibly a result of taking too many changes at once.